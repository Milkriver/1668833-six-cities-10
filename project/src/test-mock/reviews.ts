export const mockReviews = [
  {
    comment: 'The deluxe room was a quite comfortable one with all the adequate facilities. The only thing that made me feel uncomfortable was the rude behavior of an impolite staff at the reception desk.',
    date: '2022-06-23T12:25:36.939Z',
    id: 1,
    rating: 3,
    user: {
      avatarUrl: 'https://10.react.pages.academy/static/avatar/3.jpg',
      id: 12,
      isPro: true,
      name: 'Isaac'
    }
  },
  {
    comment: 'I stayed here for one night and it was an unpleasant experience.',
    date: '2022-06-23T12:25:36.939Z',
    id: 2,
    rating: 5,
    user: {
      avatarUrl: 'https://10.react.pages.academy/static/avatar/2.jpg',
      id: 11,
      isPro: false,
      name: 'Jack'
    }
  }
];
