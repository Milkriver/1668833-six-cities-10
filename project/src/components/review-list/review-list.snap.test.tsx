import { render } from '@testing-library/react';
import { mockReviews } from '../../test-mock/reviews';
import ReviewList from './review-list';

describe('Component: ReviewList', () => {
  it('should render correctly', () => {
    const { container } = render(<ReviewList comments={mockReviews} />);
    expect(container).toMatchSnapshot();
  });
});
