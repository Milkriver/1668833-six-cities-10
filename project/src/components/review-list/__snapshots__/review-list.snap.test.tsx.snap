// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ReviewList should render correctly 1`] = `
<div>
  <ul
    class="reviews__list"
  >
    <li
      class="reviews__item"
    >
      <div
        class="reviews__user user"
      >
        <div
          class="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            class="reviews__avatar user__avatar"
            height="54"
            src="https://10.react.pages.academy/static/avatar/3.jpg"
            width="54"
          />
        </div>
        <span
          class="reviews__user-name"
        >
          Isaac
        </span>
      </div>
      <div
        class="reviews__info"
      >
        <div
          class="reviews__rating rating"
        >
          <div
            class="reviews__stars rating__stars"
          >
            <span
              style="width: 60%;"
            />
            <span
              class="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          class="reviews__text"
        >
          The deluxe room was a quite comfortable one with all the adequate facilities. The only thing that made me feel uncomfortable was the rude behavior of an impolite staff at the reception desk.
        </p>
        <time
          class="reviews__time"
          datetime="2022-06-23T12:25:36.939Z"
        >
          June 2022
        </time>
      </div>
    </li>
    <li
      class="reviews__item"
    >
      <div
        class="reviews__user user"
      >
        <div
          class="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            class="reviews__avatar user__avatar"
            height="54"
            src="https://10.react.pages.academy/static/avatar/2.jpg"
            width="54"
          />
        </div>
        <span
          class="reviews__user-name"
        >
          Jack
        </span>
      </div>
      <div
        class="reviews__info"
      >
        <div
          class="reviews__rating rating"
        >
          <div
            class="reviews__stars rating__stars"
          >
            <span
              style="width: 100%;"
            />
            <span
              class="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          class="reviews__text"
        >
          I stayed here for one night and it was an unpleasant experience.
        </p>
        <time
          class="reviews__time"
          datetime="2022-06-23T12:25:36.939Z"
        >
          June 2022
        </time>
      </div>
    </li>
  </ul>
</div>
`;
